electives/18660.md

```diff
@@ -0,0 +1,47 @@
+# 18-660: Oprimization
+
+| Category | Difficulty |
+|:-:       | :-:        |
+| HW       | 4          |
+| Exams    | 3          |
+| Project  | varies     |
+
+In Spring 2022, this course was taught by Prof. Qu.
+
+## Topics Covered
+
+Some of the main topics covered here are
+
+- **What is convex optimization**
+- **Types of convex problems**
+- **Dual problem formulation**
+- **KKT Conditions**
+- **Gradient Descent**
+- **Stochastic Gradient Descent**
+- **Proximal Gradient Descent**
+- **Projected Gradient Descent (aka a lot of flavors of Gradient Descent)**
+- **Barrier Problems**
+- **Non-convex Optimization**
+
+## Course logistics
+
+There are 4 HWs and 2 midterms. The HWs are challenging, but not particularly long. You have roughly 10 days to 2 weeks for each. If you start early and go to OH, they're very maneagable. 
```

Comment:
Can you explain what contents on on the homeworks and midterms? Like what type of problems, what they are testing, how to study for them?